packages/toolbox-core/src/toolbox_core/tool.py

```diff
@@ -98,13 +94,23 @@ def __init__(
         # map of parameter name to value (or callable that produces that value)
         self.__bound_parameters = bound_params
 
+    @staticmethod
+    def _schema_to_docstring(schema: ToolSchema) -> str:
+        """Convert a tool schema into its function docstring"""
+        docstring = schema.description
+        if not schema.parameters:
+            return docstring
+        docstring += "\n\nArgs:"
+        for p in schema.parameters:
+            docstring += f"\n    {p.name} ({p.type}): {p.description}"
```

Comment:
Optional: Google-style docstring seems to put the actual class type name (eg. `str`) in place of the name of the type (eg. `"string"`) [[screenshot](https://screenshot.googleplex.com/7ADXufZLvpU45xm)]

```suggestion
            docstring += f"\n    {p.name} ({p.__get_type().__name__}): {p.description}"
```